community/kernel/src/main/java/org/neo4j/kernel/api/index/IndexProvider.java

```diff
@@ -90,32 +93,10 @@
  * {@link #getOnlineAccessor(long, IndexDescriptor, IndexSamplingConfig) online accessor} to
  * write to the index.
  */
-public abstract class SchemaIndexProvider extends LifecycleAdapter implements Comparable<SchemaIndexProvider>
+public abstract class IndexProvider<DESCRIPTOR extends IndexDescriptor> extends LifecycleAdapter implements Comparable<IndexProvider<?>>
```

Comment:
`Comparable<IndexProvider<?>>` ? maybe `Comparable<IndexProvider<DESCRIPTOR>>` instead?

Reply:
I'm not sure, I think you should be able to compare different types of index providers?